File: e2e/factories/githubIntegrations.ts
import { db } from "@test/db";
import { companiesFactory } from "@test/factories/companies";
import { integrations } from "@/db/schema";
import { assert } from "@/utils/assert";

export const githubIntegrationsFactory = {
  create: async (overrides: Partial<typeof integrations.$inferInsert> = {}) => {
    const [integration] = await db
      .insert(integrations)
      .values({
        type: "GithubIntegration",
        accountId: "1855287",
        companyId: overrides.companyId || (await companiesFactory.create()).company.id,
        configuration: {
          access_token: "gho_aGogSNKXswlREuTd3NLISAMPLE",
          organizations: ["antiwork"],
          webhooks: [{ id: "1234567890", organization: "antiwork" }],
        },
        ...overrides,
      })
      .returning();
    assert(integration != null);

    return { integration };
  },
};

File: apps/rails/spec/factories/github_integrations.rb
# frozen_string_literal: true

FactoryBot.define do
  factory :github_integration, parent: :integration, class: "GithubIntegration" do
    type { "GithubIntegration" }
    account_id { "1855287" }
    configuration do
      {
        access_token: "gho_aGogSNKXswlREuTd3NLISAMPLE",
        organizations: ["antiwork"],
        webhooks: [{ id: "1234567890", organization: "antiwork" }],
      }
    end
  end
end
